### ISSUE TITLE
Settings sync stopped working after latest update Request ... failed because of Unauthorized (401)

### ISSUE THREAD

**Issue by IllusionMH** [microsoft/vscode#114231]
Issue Type: <b>Bug</b>

I use Settings sync with Microsoft account. Everything worked fine previously, however few hours ago I started to receive notification that sync is suspended.

I tried to turn in on - same message.
Signed out and signed in to Microsoft account - same error.

Notification:
```
Settings sync is suspended because of successive authorization failures. Please sign in again to continue synchronizing
```

<details>
<summary>Output: Log (Settings Sync)</summary>

```log
[2021-01-13 00:21:34.793] [userDataSync] [info] Auto Sync: Started
[2021-01-13 00:21:34.793] [userDataSync] [info] Auto Sync: Triggered by Interval
[2021-01-13 00:21:34.859] [userDataSync] [info] Request failed {"url":"https://vscode-sync-insiders.trafficmanager.net/v1/manifest","status":401,"execution-id":"2b49b00f-ce1b-44eb-b42e-4c2c13aa02bf","operation-id":"7aa0ef42a17f4c849e57d933cf86705c"}
[2021-01-13 00:21:34.859] [userDataSync] [info] Auto Sync: Suspended until auth token is available.
[2021-01-13 00:21:34.859] [userDataSync] [info] Auto sync: Cancelled sync that is in progress
[2021-01-13 00:21:34.859] [userDataSync] [info] Auto Sync: Stopped
[2021-01-13 00:21:34.859] [userDataSync] [error] Unauthorized (UserDataSyncError) syncResource:unknown operationId:7aa0ef42a17f4c849e57d933cf86705c: Request 'https://vscode-sync-insiders.trafficmanager.net/v1/manifest' failed because of Unauthorized (401).
    at y.request (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:172458)
    at async y.manifest (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:170190)
    at async s.createSyncTask (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:150531)
    at async t.doSync (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:145159)
[2021-01-13 00:28:05.650] [userDataSync] [info] Auto Sync: Started
[2021-01-13 00:28:05.650] [userDataSync] [info] Auto Sync: Triggered by Interval
[2021-01-13 00:28:05.976] [userDataSync] [info] Request failed {"url":"https://vscode-sync-insiders.trafficmanager.net/v1/manifest","status":401,"execution-id":"6e5a00f1-7917-4d0e-beac-46a11dfaecbe","operation-id":"8bf208c9b8464eb6b2be8c2ea64bffd6"}
[2021-01-13 00:28:05.976] [userDataSync] [info] Auto Sync: Suspended until auth token is available.
[2021-01-13 00:28:05.976] [userDataSync] [info] Auto sync: Cancelled sync that is in progress
[2021-01-13 00:28:05.976] [userDataSync] [info] Auto Sync: Stopped
[2021-01-13 00:28:05.976] [userDataSync] [error] Unauthorized (UserDataSyncError) syncResource:unknown operationId:8bf208c9b8464eb6b2be8c2ea64bffd6: Request 'https://vscode-sync-insiders.trafficmanager.net/v1/manifest' failed because of Unauthorized (401).
    at y.request (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:172458)
    at async y.manifest (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:170190)
    at async s.createSyncTask (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:150531)
    at async t.doSync (file:///C:/Users/andrii_dieiev/AppData/Local/Programs/Microsoft VS Code Insiders/resources/app/out/vs/code/electron-browser/sharedProcess/sharedProcessMain.js:47:145159)
```
</details>
<details>
<summary>Output: Microsoft Authentication </summary>

timestamps looks off by 2 hours, but are recent

```log
[Info  - 22:19:30.600] Refreshing token...
[Info  - 22:19:31.256] Token refresh success
[Info  - 22:19:31.269] Token available from cache, expires in 3598987 milliseconds
[Info  - 22:19:31.282] Token available from cache, expires in 3598974 milliseconds
[Info  - 22:19:48.450] Token available from cache, expires in 3581806 milliseconds
[Info  - 22:19:53.751] Token available from cache, expires in 3576505 milliseconds
[Info  - 22:19:55.462] Token available from cache, expires in 3574794 milliseconds
[Info  - 22:19:55.851] Token available from cache, expires in 3574405 milliseconds
[Info  - 22:19:55.924] Token available from cache, expires in 3574332 milliseconds
[Info  - 22:20:03.973] Token available from cache, expires in 3566283 milliseconds
[Info  - 22:20:09.782] Logging out of session 'f8cdef31-a31e-<redacted>/1:live.com:0003BF<redacted>/761b7ba9-<redacted>'
[Info  - 22:20:19.288] Logging in...
[Info  - 22:21:34.417] Exchanging login code for token
[Info  - 22:21:34.787] Exchanging login code for token success
[Info  - 22:21:34.789] Login successful
[Info  - 22:21:34.789] Token available from cache, expires in 3598998 milliseconds
[Info  - 22:21:34.791] Token available from cache, expires in 3598996 milliseconds
[Info  - 22:21:34.791] Token available from cache, expires in 3598996 milliseconds
[Info  - 22:21:34.791] Token available from cache, expires in 3598996 milliseconds
[Info  - 22:21:34.860] Token available from cache, expires in 3598927 milliseconds
[Info  - 22:27:58.781] Token available from cache, expires in 3215006 milliseconds
[Info  - 22:28:03.394] Token available from cache, expires in 3210393 milliseconds
[Info  - 22:28:05.648] Token available from cache, expires in 3208139 milliseconds
[Info  - 22:28:05.977] Token available from cache, expires in 3207810 milliseconds
[Info  - 22:37:52.257] Token available from cache, expires in 2621530 milliseconds
[Info  - 22:37:54.106] Token available from cache, expires in 2619681 milliseconds
[Info  - 22:37:54.390] Token available from cache, expires in 2619397 milliseconds
[Info  - 22:38:47.637] Token available from cache, expires in 2566150 milliseconds
[Info  - 22:38:51.974] Token available from cache, expires in 2561813 milliseconds
[Info  - 22:38:52.49] Token available from cache, expires in 2561738 milliseconds
```
</details>

VS Code version: Code - Insiders 1.53.0-insider (e17aea136d32fdb06d9b32184b04ccd36bb51e9c, 2021-01-12T08:23:24.929Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (12 x 3192)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.72GB (16.86GB free)|
|Process Argv|--crash-reporter-id cf52dbde-2c94-45c3-9bf3-45d8b9966b04|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (26)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-css-modules|and|1.2.3
markdown-preview-github-styles|bie|0.1.6
vscode-css-modules|cli|0.2.3
bracket-pair-colorizer-2|Coe|0.2.0
vscode-svgviewer|css|2.0.0
vscode-eslint|dba|2.1.14
gitlens|eam|11.1.3
toggle-excluded-files|eam|1.7.0
EditorConfig|Edi|0.16.4
LogFileHighlighter|emi|2.10.0
vscode-pull-request-github-insiders|Git|2021.1.16382
vscode-test-explorer|hbe|2.19.4
rest-client|hum|0.24.4
vscode-peacock|joh|3.9.1
remote-ssh|ms-|0.62.0
remote-ssh-edit|ms-|0.62.0
remote-wsl|ms-|0.52.0
azure-account|ms-|0.9.4
hexeditor|ms-|1.3.0
js-debug-nightly|ms-|2021.1.817
debugger-for-edge|msj|1.0.15
material-icon-theme|PKi|4.5.0
copy-relative-path-posix|rss|0.0.4
vscode-jest-test-adapter|rtb|0.2.0
quokka-vscode|Wal|1.0.335
markdown-pdf|yza|1.4.4

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368cf:30146710
vsreu685:30147344
openlogontheside:30221882
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30208930
wsl2prompt:30219162
vstry914:30244316
pythonvsdeb440:30224570
unusedpromptcf:30219165
folderexplorer:30219166
openfilemenu:30219168
pythonvsded773:30223139
core-portspanelcf:30233468

```

</details>

**Comment by uschen** [comment]
```
{"url":"https://vscode-sync.trafficmanager.net/v1/manifest","status":401 ...}

```

**Comment by IllusionMH** [comment]
Looks similar to #99219 (thanks bot, apparently that is not marked as related to settings sync and not found by "sync 401")

@RMacfarlane is there any info that I could provide to help investigation?

Worked fine until recent hour or two. No changes to MS account recently.

**Comment by SharakPL** [comment]
Same here. Everything worked just fine about 2 hours ago. There was no updates of any kind. Code simply started to show sync failure notification, though it seems I am still logged in because I have the option to log out. Maybe a server problem...

**Comment by spicefans** [comment]
Same here, it started like 3 hours ago. No updates no nothing

**Comment by RMacfarlane** [comment]
Looking at the log for authentication, the access token was refreshed, then repeatedly requested when settings sync failed, so I don't see anything unexpected there. @Tyriar how does the setting sync server validate the access token it receives?

**Comment by the-exodus** [comment]
It appears to be some kind of concurrency-related issue. Quite often two syncs appear to be invoked at the same time, Once because of activity, and once because of interval. The interval sync appears to somehow stay queued until another sync gets scheduled, at which time both launches but only the interval succeeds.

**Comment by kavehjamshidi** [comment]
Confirming this issue. Sigining out and in doesn't fix it.

**Comment by CodeCakePie** [comment]
Same here... Using Code on MacOS and logged in with a Microsoft Account.

**Comment by huanguolin** [comment]
Same here...
![image](https://user-images.githubusercontent.com/12430413/104415222-f5213300-55b4-11eb-8350-c71225cbc26e.png)

**Comment by LilHang** [comment]
same here using insider version with Microsoft account

**Comment by Unknow-max** [comment]
Same here guys. Stopped working some hours ago.

**Comment by distante** [comment]
Kind of glad to know is not a problem with my account.

**Comment by Unknow-max** [comment]
Are we all with ms accounts logged right? No github accounts here?

**Comment by kotsias** [comment]
Same here. Stopped working today (13/01/2021).

**Comment by soundnotation** [comment]
Same here. It worked fine until this morning. The error notification appears again minutes after I signed in with my account. I have the same settings as @huanguolin except for Version: 1.52.1 (**user** setup) and I'm currently using Remote Explorer, if that matters.

**Comment by Neutrino-Sunset** [comment]
Had the same issue when using Microsoft account.

Closed down VsCode.
Removed all vscode credentials from Windows Credentials Manager.
Restarted VsCode.
Used Settings Sync to sign in using Github account.
Now seems to be working fine.

**Comment by doshongit** [comment]
Same problem here with a Microsoft account. Started this morning (GMT time). Yesterday, I had (the only time I've done this) 2 instances of VS Code open on separate machines with the same (Google Drive) folder open. Thought it may be to do with that, but the number of other people all experiencing this at roughly the same time suggests a Microsoft server-side issue.

**Comment by Gliptal** [comment]
Same here.

**Comment by gjirm** [comment]
Same for me.

**Comment by henkkelder** [comment]
Same here

**Comment by sandy081** [comment]
> Signed out and signed in to Microsoft account - same error.

@RMacfarlane It seems that the token that is provided by the authentication provider after re-signing is not getting authorized by the settings sync server. We have to see if this is token issue or server issue.

@lszomoru @Tyriar FYI

**Comment by dippas** [comment]
Same here

**Comment by nooobcoder** [comment]
Seems to work now! This thing got my nerves on. Was so tensed as the sync was failing.
Settings sync works so seemlessly, other IDEs fail to achieve this.

**Comment by Ivan-Navolotski** [comment]
Same here guys. Stopped working some hours ago.

**Comment by rodrigoguariento** [comment]
Same here. 

Version: 1.52.1
Commit: ea3859d4ba2f3e577a159bc91e3074c5d85c0523
Date: 2020-12-16T16:32:10.090Z
Electron: 9.3.5
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
**OS: Linux x64 5.8.0-36-generic snap**

**Comment by lostcodder** [comment]
Same trouble. It started from yesterday. (Microsoft account). Auth by GitHub is works but all my synced stuff at MS

**Comment by d2inco** [comment]
same here; started this morning.  same version (1.52.1, User setup), Windows.

**Comment by SoccerField24x7** [comment]
Also using a Microsoft account, but I noticed the issue last night - at approximately 23:00 GMT-8.

**Comment by jonconwayuk** [comment]
I have the same issue from this morning. Ongoing at 15:10 GMT.

**Comment by MiDevel** [comment]
Same issue, started today. Windows, v.1.52.1

**Comment by lszomoru** [comment]
Sorry about that. We are looking at the problem and will update this issue with our findings.

**Comment by lszomoru** [comment]
We have rolled out a configuration change across all regions to mitigate the issue. I have validated that I was able to enable settings sync after signing in with my Microsoft account in the stable version of VS Code. Everyone impacted, could please sign out/sign in and try to enable settings sync and let me know if you are still seeing the error. Apologies for the inconvenience.

**Comment by aheckler** [comment]
Works for me now! TY @lszomoru :)

**Comment by gjirm** [comment]
It works for me too. Thanks!

**Comment by IllusionMH** [comment]
@lszomoru now it works on same PC that initial report was from. Thanks! 🎉 
Version: 1.53.0-insider (user setup)
Commit: e17aea136d32fdb06d9b32184b04ccd36bb51e9c
Date: 2021-01-12T08:23:24.929Z
OS: Windows_NT x64 10.0.19041

Should I close this one or it should stay open for situation monitoring?

**Comment by soundnotation** [comment]
Seems to work now after signing out and in again. Cheers

**Comment by tjehlar** [comment]
It is working for me now.

**Comment by lszomoru** [comment]
Since we have not received any new reports after we have rolled out the configuration change to mitigate the problem, I will go ahead and close this issue. We have reviewed the service code/telemetry and will make some immediate improvements to both the service, and our monitoring system to make sure that such issues do not occur in the future and in case it does, it is timely detected. Apologies for the inconvenience.
